pkg/objstore/objstore.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package objstore;
+
+/* Design questions
+  - Do we want to be able to upload only part of an object
+  - What are we doing for permissions / ACLs
+  - What do we want to do for location, e.g., regions?
+  - How do object versions work?
+  - What concurrency models / consistency guarantees do we support
+  - How are we encoding error codes? Can we use standard rRPC status (https://github.com/grpc/grpc/blob/master/doc/statuscodes.md)
+*/
+
+
+service ObjectStore {
+    rpc createBucket (CreateBucketRequest) returns (Empty) {}
+    rpc listBucket (ListBucketRequest) returns (ListBucketResponse) {}
+    rpc get (GetRequest) returns (GetResponse) {}
+    rpc put (PutRequest) returns (Empty) {}
+    //    rpc deleteBucket
+}
+
+message CreateBucketRequest {
+    string bucketName = 1;
+    // TODO add permissions
+    // TODO add location
+}
+
+message ListBucketRequest {
+    string bucketName = 1;
+}
+
+message ListBucketResponse {
+    repeated string objectName = 1;
+}
+
+message GetRequest {
+    string bucketName = 1;
+    string objectName = 2;
+}
+
+message GetResponse {
+    bytes data = 1;
+}
+
+message PutRequest {
+    string bucketName = 1;
+    string objectName = 2;
+    bytes data = 3;
+}
+
+message Empty {}
```

Comment:
Nit: This shouldn't need to be redefined. You can use `import "google/protobuf/empty.proto";` and then return `google.protobuf.Empty`. There are a bunch of commonly used protobuf definitions in that package, which you can find [here](https://github.com/protocolbuffers/protobuf/tree/master/src/google/protobuf).

Reply:
Ok, let's try that. I was going to go that way but then figured this one is actually simpler to define than to import. But maybe some of the other definitions are useful.